keras_contrib/layers/normalization.py

```diff
@@ -1,3 +1,236 @@
+            if your input tensor has shape (samples, channels, rows, cols),
+            set axis to 1 to normalize per feature map (channels axis).
+        momentum: momentum in the computation of the
+            exponential average of the mean and standard deviation
+            of the data, for feature-wise normalization.
+        r_max_value: Upper limit of the value of r_max.
+        d_max_value: Upper limit of the value of d_max.
+        t_delta: At each iteration, increment the value of t by t_delta.
+        weights: Initialization weights.
+            List of 2 Numpy arrays, with shapes:
+            `[(input_shape,), (input_shape,)]`
+            Note that the order of this list is [gamma, beta, mean, std]
+        beta_init: name of initialization function for shift parameter
+            (see [initializations](../initializations.md)), or alternatively,
+            Theano/TensorFlow function to use for weights initialization.
+            This parameter is only relevant if you don't pass a `weights` argument.
+        gamma_init: name of initialization function for scale parameter (see
+            [initializations](../initializations.md)), or alternatively,
+            Theano/TensorFlow function to use for weights initialization.
+            This parameter is only relevant if you don't pass a `weights` argument.
+        gamma_regularizer: instance of [WeightRegularizer](../regularizers.md)
+            (eg. L1 or L2 regularization), applied to the gamma vector.
+        beta_regularizer: instance of [WeightRegularizer](../regularizers.md),
+            applied to the beta vector.
+
+    # Input shape
+        Arbitrary. Use the keyword argument `input_shape`
+        (tuple of integers, does not include the samples axis)
+        when using this layer as the first layer in a model.
+
+    # Output shape
+        Same shape as input.
+
+    # References
+        - [Batch Normalization: Accelerating Deep Network Training by Reducing Internal Covariate Shift](https://arxiv.org/abs/1502.03167)
+    """
+
+    def __init__(self, epsilon=1e-3, mode=0, axis=-1, momentum=0.99,
+                 r_max_val=3., d_max_val=5., t_delta=1., weights=None, beta_init='zero',
+                 gamma_init='one', gamma_regularizer=None, beta_regularizer=None,
+                 **kwargs):
+        self.supports_masking = True
+        self.beta_init = initializations.get(beta_init)
+        self.gamma_init = initializations.get(gamma_init)
+        self.epsilon = epsilon
+        self.mode = mode
+        self.axis = axis
+        self.momentum = momentum
+        self.gamma_regularizer = regularizers.get(gamma_regularizer)
+        self.beta_regularizer = regularizers.get(beta_regularizer)
+        self.initial_weights = weights
+        self.r_max_value = r_max_val
+        self.d_max_value = d_max_val
+        self.t_delta = t_delta
+        if self.mode == 0 or self.mode == 2:
```

Comment:
Hmm it is true that they _should_ be same at inference time, since clipping should be minimal at the end when r and d are max, for some reason, no matter what value I used for t_delta, I could not properly train a Renorm model for semantic segmentation. 

From the very beginning, the dice_score just oscillates between 0.4 and 0.47. The moment I switched to the Mode 2 on the older version (with the std / variance problem fixed), the score leaped as expected from 0.4 to 0.76 in just 8 epochs. After that it rose slowly, to a final value of about 0.84 or so after 100 epochs. The training images were definitely very difficult to train, but it still got a good score considering the model was quite small. 

For reference, with batchnorm mode 0, it was stuck at 0.2-0.34. With mode 2, it rose to a maximum of 0.8293 after 100 epochs. This means that unlike batchnorm mode 0 which at testing time uses moving statistics, BRN is actually working correctly, since at mode 0 (which merges mode 0 and 2 of BN), it is still learning, just not well enough. 

I agree that it is definitely a model technique, not described in the paper. I will leave it to your judgement to decide if it should be included.